src/commands/crashlytics-mappingfile-upload.ts

```diff
@@ -0,0 +1,73 @@
+import { Command } from "../command";
+import { FirebaseError } from "../error";
+import * as utils from "../utils";
+
+import { fetchBuildtoolsJar, runBuildtoolsCommand } from "../crashlytics/buildToolsJarHelper";
+import { Options } from "../options";
+
+interface CommandOptions extends Options {
+  app?: string;
+  mappingFile?: string;
+  resourceFile?: string;
+}
+
+interface JarOptions {
+  app: string;
+  mappingFilePath: string;
+  resourceFilePath: string;
+}
+
+export const command = new Command("crashlytics:mappingfile:upload <mappingFile>")
+  .description("upload a ProGuard/R8-compatible mapping file to deobfuscate stack traces")
+  .option("--app <appID>", "The app id of your Firebase app")
+  .option(
+    "--resource-file <resourceFile>",
+    "Path to the Android resource XML file that includes the mapping file id."
+  )
+  .option("--debug", "print debug output and logging from the underlying uploader tool")
+  .action(async (mappingFile: string, options: CommandOptions) => {
+    const app = getGoogleAppID(options);
+    const debug = !!options.debug;
+    if (!mappingFile) {
+      throw new FirebaseError(
+        "set <mappingFile> to a valid mapping file path, e.g. app/build/outputs/mapping.txt"
```

Comment:
Should this maybe specify the flag rather than the variable? Below would match what's in generateid:

```suggestion
        "set `--mapping-file <mappingFile>` to a valid mapping file path, e.g. app/build/outputs/mapping.txt"
```